apps/tasks.py

```diff
@@ -649,3 +505,142 @@ def delete_old_objects():
+                    "status": status,
+                }
+                status_object = AppStatus(appinstance=appinstance)
+                update_status(appinstance, status_object, status)
+    except Exception as exc:
+        # Catch other exceptions to trigger a retry
+        raise self.retry(exc=exc)
+
+
+@worker_ready.connect
+def on_worker_ready(**kwargs):
+    """
+    This function starts the event listener in a given namespace.
+    When the Celery worker is ready, the task is started.
+    """
+    label_selector = "type=app"
+    NAMESPACE = settings.NAMESPACE
+    sync_all_statuses(namespace=NAMESPACE, label_selector=label_selector)
+    init_event_listener.apply_async(args=(NAMESPACE, label_selector), countdown=1)
+
+
+def setup_client():
+    """
+    Sets up the kubernetes python client and event streamer
+    """
+    try:
+        config.load_incluster_config()
+    except config.ConfigException:
+        try:
+            config.load_kube_config(settings.KUBECONFIG)
+        except config.ConfigException:
+            raise config.ConfigException(
+                "Could not set the cluster config. Try to use the cluster.conf file or set incluster config"
+            )
+
+    api = client.CoreV1Api()
+    w = watch.Watch()
+    return api, w
+
+
+def get_status(pod):
+    """
+    Returns the status of a pod by looping through each container
+    and checking the status.
+    """
+    container_statuses = pod.status.container_statuses
+
+    if container_statuses is not None:
+        for container_status in container_statuses:
+            state = container_status.state
+
+            if state is not None:
+                terminated = state.terminated
+
+                if terminated is not None:
+                    return terminated.reason
+
+                waiting = state.waiting
+
+                if waiting is not None:
+                    return waiting.reason
+
+                running = state.running
+
+                if running is not None:
+                    return "Running"
+
+            print("Last state not found.")
+
+    return pod.status.phase
+
+
+def sync_all_statuses(namespace, label_selector):
+    """
+    Syncs the status of all apps with a pod that is on the cluster
+    """
+    api, _ = setup_client()
+    for pod in api.list_namespaced_pod(namespace=namespace, label_selector=label_selector).items:
+        status = pod.status.phase
+        release = pod.metadata.labels["release"]
+        appinstance = AppInstance.objects.filter(parameters__contains={"release": release}).last()
+        if appinstance:
+            status_object = AppStatus(appinstance=appinstance)
+            update_status(appinstance, status_object, status)
+
+
+def update_status(appinstance, status_object, status):
+    """
+    Helper function to update the status of an appinstance and a status object.
+    """
+    status_object.status_type = status
+    status_object.save()
```

Comment:
Please look into using argument update_fields= to restrict which fields get updated, especially since I think you mentioned that there are cases where 2 update (save) processes step on each other and the latest update overwrites the first.
https://docs.djangoproject.com/en/4.2/ref/models/instances/#ref-models-update-fields

Reply:
Thanks! I have now updated according to your feedback